### DISCUSSION TITLE
Support for concurernt processing workloads (e.g. with fibers)

### DISCUSSION THREAD

**Discussion by Rastusik** [getsentry/sentry-php#1473] [Q&A]
### Problem Statement

The current version of SDK contains static calls to classes that manage one global state. This is not usable in concurrent workloads, where multiple fibers may process multiple requests concurrently.

Would it be possible to implement a fiber context layer above the current global state, so each fiber could work with contextual data that belongs to it?

### Solution Brainstorm

Maybe it would be sufficient to index fiber contexts by `spl_object_id()` of the fiber, not sure if that works though. In Swoole, the contexts can be indexed by coroutine id.

**Comment by cleptric** [comment]
Thanks for letting us know about this. This is not on the top list of our priorities right now, but we may look at a better way to support concurrent PHP runtimes at a later date.

**Comment by huangdijia** [comment]
@Rastusik You can try Hyperf, a collaborative framework based on Swoole. I extended sentry based on Hyperf.

https://github.com/friendsofhyperf/sentry
